harness/determined/pytorch/_pytorch_context.py

```diff
@@ -57,4 +85,311 @@ def get_lr_scheduler(self) -> Optional[pytorch.LRScheduler]:
             * ``optimizer()``
             * ``create_lr_scheduler()``
         """
-        return self.lr_scheduler
+        # TODO(DET-3267): deprecate this when releasing pytorch flexible primitives.
+        check.lt_eq(len(self.lr_schedulers), 1)
+        if len(self.lr_schedulers) == 1:
+            return self.lr_schedulers[0]
+        return None
+
+    def _Model(self, model: torch.nn.Module) -> torch.nn.Module:
+        check.is_none(self.model, "Must call Model() only once and before calling configure_amp.")
+        self.model = model.to(self.device)
+        if not self.hvd_config.use and self.n_gpus > 1:
+            check.eq(
+                self.hvd_config.aggregation_frequency,
+                1,
+                "Please enable `optimized_parallel` to use aggregation "
+                "frequency greater than 1 for single machine multi-GPU "
+                "training.",
+            )
+            self.model = nn.DataParallel(self.model)
+            logging.debug("Initialized mode for native parallel training.")
+        return model
+
+    def _Optimizer(self, optimizer: torch.optim.Optimizer) -> torch.optim.Optimizer:  # type: ignore
+        check.is_not_none(self.model, "Must have one model first.")
```

Comment:
non-blocking: this message makes it sound like users may specify more than model.